ext/ffi/00_ffi.js

```diff
@@ -13,15 +15,34 @@
       this.#rid = core.opSync("op_ffi_load", { path, symbols });
 
       for (const symbol in symbols) {
-        this.symbols[symbol] = symbols[symbol].nonblocking
-          ? (...parameters) =>
+        const isNonBlocking = symbols[symbol].nonblocking;
+
+        this.symbols[symbol] = (...args) => {
+          const parameters = [];
+          const buffers = [];
+
+          for (const arg of args) {
+            if (
+              arg?.buffer instanceof ArrayBuffer &&
+              arg.byteLength !== undefined
+            ) {
+              parameters.push(buffers.length);
+              buffers.push(arg);
+            } else {
+              parameters.push(arg);
+            }
+          }
+
+          if (isNonBlocking) {
             core.opAsync("op_ffi_call_nonblocking", {
               rid: this.#rid,
               symbol,
               parameters,
-            })
-          : (...parameters) =>
+            });
+          } else {
```

Comment:
`return`

Are we missing a test for this?

Reply:
We do. I also forgot to forward buffers, fixed now.